### PR TITLE
Generate env.rake and load all tasks in lib/tasks

### LINKED ISSUE #210: Default Rake task: environment
### Description

Add support for a default Rake task `environment`. This should serve as a base for user defined rake tasks.

Real world example:

```ruby
namespace :storage do
  desc 'Clear local storage'
  task clear: :environment do
    # Here I want to have access to the objects of my application
    # Example:
    BookRepository.clear
  end
end
```

### Implementation hints

```ruby
task :environment do
  require 'lotus/environment'
  require Lotus::Environment.new.env_config
end
```

This article can be helpful for testing this feature: https://robots.thoughtbot.com/test-rake-tasks-like-a-boss

### PR DESCRIPTION
**Pull request by joneslee85**
[Resolve #210]

### CODE REVIEW

File: lib/lotus/generators/application/app.rb [hunk 1]
```diff
@@ -55,7 +56,7 @@ def start
             "lib/#{ app_name }/entities",
             "lib/#{ app_name }/repositories",
             "public/javascripts",
-            "public/stylesheets"
+            "public/stylesheets",
```
Comment (AlfonsoUceda) [inline lib/lotus/generators/application/app.rb]: unnecessary coma

File: lib/lotus/generators/application/app.rb [hunk 2]
```diff
@@ -47,6 +47,7 @@ def start
             'config/routes.rb.tt'               => 'config/routes.rb',
             'views/application_layout.rb.tt'    => 'app/views/application_layout.rb',
             'templates/application.html.erb.tt' => 'app/templates/application.html.erb',
+            'lib/tasks/env.rake.tt'             => 'lib/tasks/env.rake',
```
Comment (AlfonsoUceda) [inline lib/lotus/generators/application/app.rb]: unnecessary coma at the end

File: lib/lotus/generators/application/container.rb [hunk 3]
```diff
@@ -39,11 +39,12 @@ def start
             'config/environment.rb.tt' => 'config/environment.rb',
             'lib/app_name.rb.tt'       => "lib/#{ app_name }.rb",
             'lib/config/mapping.rb.tt' => 'lib/config/mapping.rb',
+            'lib/tasks/env.rake.tt'    => 'lib/tasks/env.rake',
```
Comment (AlfonsoUceda) [inline lib/lotus/generators/application/container.rb]: unnecessary coma

File: lib/lotus/generators/application/container.rb [hunk 4]
```diff
@@ -39,11 +39,12 @@ def start
...
           }
 
           empty_directories = [
             "lib/#{ app_name }/entities",
-            "lib/#{ app_name }/repositories"
+            "lib/#{ app_name }/repositories",
```
Comment (AlfonsoUceda) [inline lib/lotus/generators/application/container.rb]: unnecessary coma

File: test/integration/rake_test.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  after do
+    Dir.chdir @current_dir
+    @current_dir = nil
+  end
+
+  it 'inherits environment and returns environment correctly' do
+    output = `bundle exec rake inspect`.strip
+    output.must_equal 'correctly inherited'
```
Comment (AlfonsoUceda) [inline test/integration/rake_test.rb]: you should assert environment values

File: test/fixtures/rake_test_app/config.ru [hunk 6]
```diff
@@ -0,0 +1,5 @@
+require_relative 'apps/backend/application'
+
+run Lotus::Router.new {
```
Comment (AlfonsoUceda) [inline test/fixtures/rake_test_app/config.ru]: use `do`` instead `{`

File: test/fixtures/rake_test_app/config.ru [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+  mount Backend::Application,  at: '/'
```
Comment (AlfonsoUceda) [inline test/fixtures/rake_test_app/config.ru]: remove the block, this is done in configure

**Comment by AlfonsoUceda** [comment]
:+1:

File: lib/lotus/generators/application/app.rb [hunk 8]
```diff
@@ -47,6 +47,7 @@ def start
             'config/routes.rb.tt'               => 'config/routes.rb',
             'views/application_layout.rb.tt'    => 'app/views/application_layout.rb',
             'templates/application.html.erb.tt' => 'app/templates/application.html.erb',
+            'lib/tasks/env.rake.tt'             => 'lib/tasks/env.rake'
```
Comment (brennovich) [inline lib/lotus/generators/application/app.rb]: Are you against expanding this abbr? `env => environment`, I think it'll avoid future confusions if the file has the same name of the task :)

Comment (joneslee85) [inline lib/lotus/generators/application/app.rb]: Not really, good suggestion, I'll do that

**Comment by jodosha** [comment]
@joneslee85 By generating an `env.rake` in new apps, it will create problems for Lotus maintenance.

For instance, [this uses](https://github.com/lotus/lotus/pull/290/files#diff-635b18fa1b7bca9c30d92f9f12bcb9feR3) `require Lotus::Environment.new.env_config`. If for some reason we want to rename that method with something else, we have suddenly broken apps generated with 0.5.0.

I'd say that `Rakefile` should have a line like this at the top:

```ruby
load 'lotus/rake_tasks'
# or
load Lotus.rake_tasks
```

That will give us the freedom to keep framework details (eg, `env_config`) hidden within that file.

**Comment by joneslee85** [comment]
I close it to redo a different version

**Pull request closed without merging**